MobileCenterCrashes/MobileCenterCrashes/Internals/MSWrapperExceptionManager.m

```diff
@@ -1,295 +1,118 @@
-  NSString *filename = [[self class] getFilenameWithUUIDRef:uuidRef];
-  [self saveWrapperExceptionData:uuidRef];
-  BOOL success = [NSKeyedArchiver archiveRootObject:self.wrapperException toFile:filename];
-  if (!success) {
-    MSLogError([MSCrashes logTag], @"Failed to save file %@", filename);
-  }
-}
-
-- (void)deleteWrapperExceptionWithUUID:(CFUUIDRef)uuidRef {
-  NSString *path = [MSWrapperExceptionManager getFilenameWithUUIDRef:uuidRef];
-  [[self class] deleteFile:path];
-
-  if ([[self class] isCurrentUUIDRef:uuidRef]) {
-    self.currentUUIDRef = nil;
-    self.wrapperException = nil;
-  }
-}
-
-- (void)deleteAllWrapperExceptions {
-  self.currentUUIDRef = nil;
-  self.wrapperException = nil;
-
-  NSFileManager *fileManager = [NSFileManager defaultManager];
-
-  for (NSString *filePath in [fileManager enumeratorAtPath:[[self class] directoryPath]]) {
-    if (![[self class] isDataFile:filePath]) {
-      NSString *path = [[[self class] directoryPath] stringByAppendingPathComponent:filePath];
-      [[self class] deleteFile:path];
+  MSErrorReport *correspondingReport = nil;
+  for (MSErrorReport *report in reports) {
+    if ([lastSavedWrapperException.processId unsignedLongValue] == report.appProcessIdentifier) {
+      correspondingReport = report;
+      break;
     }
   }
-}
-
-- (void)saveWrapperExceptionData:(CFUUIDRef)uuidRef {
-  if (!self.unsavedWrapperExceptionData) {
-    return;
+  if (correspondingReport) {
+    [self saveWrapperException:lastSavedWrapperException withBaseFilename:correspondingReport.incidentIdentifier];
   }
-  NSString *dataFilename = [[self class] getDataFilenameWithUUIDRef:uuidRef];
-  [self.unsavedWrapperExceptionData writeToFile:dataFilename atomically:YES];
 }
 
-- (NSData *)loadWrapperExceptionDataWithUUIDString:(NSString *)uuidString {
-  NSString *dataFilename = [[self class] getDataFilename:uuidString];
-  NSData *data = self.wrapperExceptionData[dataFilename];
-  if (data) {
-    return data;
-  }
-  NSError *error = nil;
-  data = [NSData dataWithContentsOfFile:dataFilename options:NSDataReadingMappedIfSafe error:&error];
-  if (error) {
-    MSLogError([MSCrashes logTag], @"Error loading file %@: %@", dataFilename, error.localizedDescription);
-  }
-  return data;
-}
+#pragma mark Helper methods
 
-- (void)deleteWrapperExceptionDataWithUUIDString:(NSString *)uuidString {
-  NSString *dataFilename = [[self class] getDataFilename:uuidString];
-  NSData *data = [self loadWrapperExceptionDataWithUUIDString:uuidString];
-  if (data) {
-    self.wrapperExceptionData[dataFilename] = data;
-  }
-  [[self class] deleteFile:dataFilename];
-}
+/**
+ * Saves a wrapper exception to disk with the given file name.
+ */
++ (void)saveWrapperException:(MSWrapperException *)wrapperException withBaseFilename:(NSString *)baseFilename {
+  NSURL *exceptionFileURL = [self getAbsoluteFileURL:baseFilename];
+  BOOL success = [MSUtility createFileAtURL:exceptionFileURL];
+  if (success) {
```

Comment:
What if it cannot create the file? You don't need to get a return value or error logs?